experiment/selinux/src/selinux.rs

```diff
@@ -44,27 +87,196 @@ impl SELinux {
+                .unwrap_or(&String::new())
+                .to_string()
+        }
+    }
+
+    // function similar with readConfig in go-selinux repo.
+    // This function reads SELinux config file and returns the value with a specified key.
+    fn read_config(target: &str) -> String {
+        match File::open(format!("{}{}", SELINUX_DIR, SELINUX_CONFIG)) {
+            Ok(file) => {
+                let reader = BufReader::new(file);
+                for line in reader.lines().map_while(Result::ok) {
+                    if line.is_empty() {
+                        continue;
+                    }
+                    if (line.starts_with(';')) || (line.starts_with('#')) {
+                        continue;
+                    }
+                    let fields: Vec<&str> = line.splitn(2, '=').collect();
+                    if fields.len() < 2 {
+                        return String::new();
+                    }
+                    if fields[0] == target {
+                        return fields[1].to_owned();
+                    }
+                }
+                String::new()
+            }
+            Err(_) => String::new(),
+        }
     }
 
     // function similar with getEnabled in go-selinux repo.
     // get_enabled returns whether SELinux is enabled or not.
-    pub fn get_enabled() -> bool {
-        unimplemented!("not implemented yet")
+    pub fn get_enabled(&mut self) -> bool {
+        let fs = Self::get_selinux_mountpoint(self);
+        if fs == String::new() {
+            return false;
+        }
+        match Self::current_label(self) {
+            Ok(con) => {
+                if con != "kernel" {
+                    return true;
+                }
+                false
+            }
+            Err(_) => false,
+        }
+    }
+
+    // function similar with verifySELinuxfsMount in go-selinux repo.
+    // verify_selinux_fs_mount verifies if the specified mount point is
+    // properly mounted as a writable SELinux filesystem.
+    fn verify_selinux_fs_mount(mnt: &Path) -> bool {
+        loop {
+            match statfs::statfs(mnt) {
+                Ok(stat) => {
+                    // verify if the file is readonly or not
+                    if !stat.flags().contains(statvfs::FsFlags::ST_RDONLY) {
+                        return false;
+                    }
+                    // verify if the file is SELinux filesystem
+                    return stat.filesystem_type() == statfs::SELINUX_MAGIC;
+                }
+                // check again if there is an issue while calling statfs
+                Err(Errno::EAGAIN) | Err(Errno::EINTR) => continue,
+                Err(_) => return false,
+            }
+        }
+    }
+
+    // function similar with findSELinuxfsMount in go-selinux repo.
+    // find_selinux_fs_mount returns a next selinuxfs mount point found,
+    // if there is one, or an empty string in case of EOF or error.
+    fn find_selinux_fs_mount(s: &str) -> String {
```

Comment:
Is the type of the return value `PathBuf`?
```suggestion
    fn find_selinux_fs_mount(s: &str) ->Option<PathBuf> {
```